### PR TITLE
Buffer.lastIndexOf

### PR DESCRIPTION
**Pull request by dcposch**
@feross

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -643,7 +643,72 @@ Buffer.prototype.compare = function compare (target, start, end, thisStart, this
...
+  // Finally, search either indexOf (if dir is true) or lastIndexOf
+  if (Buffer.isBuffer(val)) {
+    // Special case: looking for empty string/buffer always fails
+    if (val.length === 0) {
+      return -1
+    }
+    return arrayIndexOf(buffer, val, byteOffset, encoding, dir)
+  } else if (typeof val === 'number') {
+    val = val & 0xFF // Search for a byte value [0-255]
+    if (Buffer.TYPED_ARRAY_SUPPORT && Uint8Array.prototype.indexOf === 'function') {
```
Comment (feross) [inline index.js]: Whoa. I just realized this is a bug. Should be `typeof Uint8Array.prototype.indexOf === 'function'`.

**Comment by feross** [comment]
LGTM! Thanks for doing this! I added you as a maintainer, @dcposch!

File: index.js (hunk 1 above)
Comment (feross) [inline index.js]: Fixed in https://github.com/feross/buffer/pull/128

**Comment by feross** [comment]
Released as 4.9.0.

**Comment by dcousens** [comment]
For those of us reviewing this,  what was the PR motivation?

**Comment by dcousens** [comment]
Nevermind, seen https://github.com/feross/buffer/issues/114#issuecomment-238175726

**Pull request merged**